python_modules/dagster/dagster/_core/asset_graph_view/asset_graph_view.py

```diff
@@ -0,0 +1,277 @@
+    ):
+        import pendulum
+
+        from dagster._core.definitions.data_version import CachingStaleStatusResolver
+        from dagster._core.instance import DagsterInstance
+
+        stale_resolver = CachingStaleStatusResolver(
+            instance=instance or DagsterInstance.ephemeral(),
+            asset_graph=defs.get_asset_graph(),
+        )
+        check.invariant(stale_resolver.instance_queryer, "Ensure instance queryer is constructed")
+        return AssetGraphView(
+            stale_resolver=stale_resolver,
+            temporal_context=TemporalContext(
+                effective_dt=effective_dt or pendulum.now(),
+                last_event_id=last_event_id,
+            ),
+        )
+
+    def __init__(
+        self,
+        *,
+        temporal_context: TemporalContext,
+        stale_resolver: "CachingStaleStatusResolver",
+    ):
+        # Current these properties have the ability to be lazily constructed.
+        # We instead are going to try to retain the invariant that they are
+        # constructed upfront so that initialization time is well-understood
+        # and deterministic. If there are cheap operations that do not
+        # require these instances, we should move them to a different abstraction.
+
+        # ensure it is already constructed rather than created on demand
+        check.invariant(stale_resolver._instance_queryer)  # noqa: SLF001
+        # ensure it is already constructed rather than created on demand
+        check.invariant(stale_resolver._asset_graph)  # noqa: SLF001
+
+        self._stale_resolver = stale_resolver
+        # stale resolve has a CachingInstanceQueryer which has a DagsterInstance
+        # so just passing the CachingStaleStatusResolver is enough
+        self._temporal_context = temporal_context
+
+    @property
+    def effective_dt(self) -> datetime:
+        return self._temporal_context.effective_dt
+
+    @property
+    def last_event_id(self) -> Optional[int]:
+        return self._temporal_context.last_event_id
+
+    @property
+    def asset_graph(self) -> "AssetGraph":
+        return self._stale_resolver.asset_graph
+
+    @property
+    def _queryer(self) -> "CachingInstanceQueryer":
+        return self._stale_resolver.instance_queryer
+
+    def _get_partitions_def(self, asset_key: "AssetKey") -> Optional["PartitionsDefinition"]:
+        return self.asset_graph.get_partitions_def(asset_key)
+
+    def get_asset_slice(self, asset_key: "AssetKey") -> "AssetSlice":
+        # not compute_asset_slice because dynamic partitions store
+        # is just passed to AssetSubset.all, not invoked
+        return _slice_from_subset(
+            self,
+            AssetSubset.all(
+                asset_key=asset_key,
+                partitions_def=self._get_partitions_def(asset_key),
+                dynamic_partitions_store=self._queryer,
+                current_time=self.effective_dt,
+            ),
+        )
+
+    def create_parent_asset_slice(
```

Comment:
seems like this should be `compute_parent_asset_slice` to align with the conventions in `AssetSlice`.